### PR TITLE
Node-provisioning-plugin

### PR DESCRIPTION
**Pull request by skanthed**

### CODE REVIEW

**Comment by skanthed** [comment]
@tzumainn

File: plugins/esi_node_provisioning.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        """Scenario to test 'openstack esi node provisioning' command."""
+        command = f"metalsmith deploy --resource-class baremetal --image {image} --network {network} --candidate {candidate} --ssh-public-key {sshpublickey}"
+        if image:
+            command += f" --image {image}"
+        if candidate:
+            command += f" --candidate {candidate}"
+        if network:
+            command += f" --network {network}"
+        if sshpublickey:
+            command += f" --ssh-public-key {sshpublickey}"
```
Comment (tzumainn) [inline plugins/esi_node_provisioning.py]: For readability, I'd suggest calling this parameter `ssh_public_key`

File: plugins/esi_node_provisioning.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import time
+import os
+
+@validation.add("required_platform", platform="openstack", users=True)
+@scenario.configure(name="ScenarioPlugin.esi_node_provisioning")
+
+class EsiNodeProvisioning(scenario.Scenario):
+    def esi_node_provisioning(self, image=None, network=None, candidate=None, sshpublickey=None):
+        """Scenario to test 'openstack esi node provisioning' command."""
+        command = f"metalsmith deploy --resource-class baremetal --image {image} --network {network} --candidate {candidate} --ssh-public-key {sshpublickey}"
```
Comment (tzumainn) [inline plugins/esi_node_provisioning.py]: It's probably worth parameterizing the `resource_class` parameter as well.

File: plugins/esi_node_provisioning.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            command += f" --network {network}"
+        if sshpublickey:
+            command += f" --ssh-public-key {sshpublickey}"
+
+        start_time = time.time()
+        output = os.system(command)
+        end_time = time.time()
+        execution_time = end_time - start_time
+
+    def run(self, image=None, network=None, candidate=None, sshpublickey=None):
```
Comment (tzumainn) [inline plugins/esi_node_provisioning.py]: Defaulting parameters like this should only really be done if the parameter is optional; none of these are. I might suggest:

`def run(self, image, network, candidate, ssh_public_key=None, resource_class='baremetal'):`

**Review by tzumainn — CHANGES_REQUESTED** [review]
Looks good as a whole! Some minor comments inline.

File: plugins/esi_node_provisioning.py (hunk 1 above)
Comment (skanthed) [reply plugins/esi_node_provisioning.py]: It gave an error with '-' in ssh-public-key, but I can update it with ssh_public_key.

File: plugins/esi_node_provisioning.py (hunk 3 above)
Comment (skanthed) [reply plugins/esi_node_provisioning.py]: Sure will update that.

File: plugins/esi_node_provisioning.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import time
+import os
+
+@validation.add("required_platform", platform="openstack", users=True)
+@scenario.configure(name="ScenarioPlugin.esi_node_provisioning")
+
+class EsiNodeProvisioning(scenario.Scenario):
+    def esi_node_provisioning(self, image, network, candidate, resource_class='baremetal', ssh_public_key=None):
+        """Scenario to test 'openstack esi node provisioning' command."""
+        command = f"metalsmith deploy --resource-class {resource_class} --image {image} --network {network} --candidate {candidate} --ssh-public-key {ssh_public_key}"
```
Comment (tzumainn) [inline plugins/esi_node_provisioning.py]: I think the `--ssh-public-key` flag should be removed here, since it's added below.

Comment (skanthed) [reply plugins/esi_node_provisioning.py]: Done.

**Review by tzumainn — APPROVED** [review]
Looks good - thanks!

**Pull request merged**